integration-tests/testkit/test.ts

```diff
@@ -0,0 +1,102 @@
+/**
+ * This module uses Vitest's fixture system to make common usage patterns
+ * of our testkit easily consumable in test cases. @see https://vitest.dev/guide/test-context.html#test-extend
+ */
+
+import { test as testBase } from 'vitest';
+import { CLI, createCLI } from './cli';
+import { ProjectType } from './gql/graphql';
+import { initSeed, OrgSeed, OwnerSeed, ProjectSeed, Seed, TargetAccessTokenSeed } from './seed';
+
+interface Context {
+  seed: Seed;
+  owner: OwnerSeed;
+  org: OrgSeed;
+  //
+  // "single" branch
+  //
+  projectSingle: ProjectSeed;
+  targetAccessTokenSingle: TargetAccessTokenSeed;
+  cliSingle: CLI;
+  //
+  // "federation" branch
+  //
+  projectFederation: ProjectSeed;
+  targetAccessTokenFederation: TargetAccessTokenSeed;
+  cliFederation: CLI;
+  //
+  // "stitching" branch
+  //
+  projectStitching: ProjectSeed;
+  targetAccessTokenStitching: TargetAccessTokenSeed;
+  cliStitching: CLI;
+}
+
+export const test = testBase.extend<Context>({
```

Comment:
Why this and not:

```ts
test('...', async () => {
  const project = await createProject(...);
});
```

What's the reason to provide a custom `test` function?

Reply:
- Helps us write consistent test fixtures that are centrally managed, and think harder about cases where we deviate from standard fixtures, which IMO is the right balance because deviations increase maintenance cost.

- Fixtures build upon one another. When a test brings in a fixture, the dependencies of that fixture are automatically run. The result is focused test code in majority cases while maintaining possibility for any test to begin from the base fixture whereupon it builds up resources in some custom way unique to itself.